trivy-operator.yaml

```
@@ -0,0 +1,96 @@
+  copyright:
+    - license: Apache-2.0
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/aquasecurity/trivy-operator
+      tag: v${{package.version}}
+      expected-commit: 0816542808e81dc318a34a1ce8a5cd004c59dfdd
+
+  - uses: go/bump
+    with:
+      deps: |-
+        github.com/containerd/containerd@v1.7.27
+        github.com/containerd/containerd/v2@v2.0.4
+        github.com/golang-jwt/jwt/v4@v4.5.2
+        github.com/golang-jwt/jwt/v5@v5.2.2
+        golang.org/x/oauth2@v0.27.0
+        helm.sh/helm/v3@v3.17.3
+        golang.org/x/net@v0.38.0
+
+  - uses: go/build
+    with:
+      packages: "./cmd/trivy-operator"
+      output: trivy-operator
+      ldflags: |-
+        -X main.version=${{package.version}}
+        -X main.commit=$(git rev-parse HEAD)
+        -X main.date=$(date -u -d "@$SOURCE_DATE_EPOCH" +"%Y-%m-%dT%H:%M:%SZ")
+
+subpackages:
+  - name: ${{package.name}}-compat
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/usr/local/bin
+          ln -sf /usr/bin/trivy-operator ${{targets.contextdir}}/usr/local/bin
+    description: Compat package for ${{package.name}}
+    test:
+      pipeline:
+        - runs: |
+            stat /usr/local/bin/trivy-operator
+
+update:
+  enabled: true
+  github:
+    identifier: aquasecurity/trivy-operator
+    strip-prefix: v
+
+test:
+  environment:
+    environment:
+      OPERATOR_NAMESPACE: "trivy-system"
+    contents:
+      packages:
+        - ${{package.name}}
+  pipeline:
+    - name: "Basic binary verification"
+      runs: |
+        # Check if binary produces expected startup message (script exits if grep fails)
+        trivy-operator version 2>&1 | grep -q "Starting operator"
+
+        # Verify package version in ldflags
+        if ! strings /usr/bin/trivy-operator | grep -q "version=${{package.version}}"; then
+          echo "Error: Package version ${{package.version}} not found in binary"
+          exit 1
+        fi
+
+        echo "Basic binary verification passed"
+    - uses: test/kwok/cluster
+    - uses: test/daemon-check-output
+      with:
+        timeout: 60
+        setup: |
+          # Create trivy-system namespace
+          kubectl create namespace trivy-system
+
+          # Create required configmap
+          kubectl create configmap trivy-operator -n trivy-system \
+            --from-literal=trivy.ignoreUnfixed=true \
+            --from-literal=trivy.skipFiles=/etc/ssl/certs/ca-certificates.crt \
+            --from-literal=trivy.skipDirs=/usr/lib/ruby/gems/2.7.0/cache \
+            --from-literal=trivy.insecure=false \
+            --from-literal=trivy.offlineScan=false \
+            --from-literal=trivy.securityChecks=vuln,secret \
+            --from-literal=trivy.severity=UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL \
+            --from-literal=trivy.timeout=5m0s
+        start: /usr/bin/trivy-operator
```

```suggestion
        start: trivy-operator
```
Upstream image contents:
```
            "WorkingDir": "/",
            "Entrypoint": [
                "trivy-operator"
            ]
```
More context here: https://github.com/wolfi-dev/os/pull/50254/files#r2068524111

---

Under the hood, Melange’s go/build step puts your compiled binary into usr/bin/<output> by default (that is, it uses prefix=usr and install-dir=bin), so our trivy-operator lands at /usr/bin/trivy-operator. When you run melange test, the harness drops you into a shell whose $PATH already includes /usr/bin, which means typing trivy-operator automatically runs the /usr/bin version, so we can write anything that makes the tests clearer,

check out the go/build pipeline definition here for more understanding.
https://github.com/chainguard-dev/melange/blob/main/pkg/build/pipelines/go/build.yaml

---

@rahulduvedi  I'm aware and already mentioned the same here: https://github.com/wolfi-dev/os/pull/50254/files#r2068524111

It's generally recommended to go match tests for upstream image. I'm fine with either approach.